homeassistant/components/maxcube/config_flow.py

```diff
@@ -0,0 +1,62 @@
+"""Adds config flow for maxcube."""
+from maxcube.cube import MaxCube
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_SCAN_INTERVAL
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.util.dt import now
+
+from . import DEFAULT_PORT, DOMAIN
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class MaxCubeFlowHandler(config_entries.ConfigFlow):
+    """Config flow for MaxCube."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize."""
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if user_input is not None:
+            host = user_input.get(CONF_HOST)
+            port = user_input.get(CONF_PORT, DEFAULT_PORT)
+            title = f"{host}"
+            if port != DEFAULT_PORT:
+                title += f":{port}"
+
+            try:
+                cube = MaxCube(host, port, now=now)
+                cube.disconnect()
+            except Exception:  # pylint: disable=broad-except
+                return self.async_abort(reason="connection_error")
+
+            return self.async_create_entry(title=title, data=user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                    vol.Optional(CONF_PORT, default=DEFAULT_PORT): int,
+                    vol.Optional(CONF_SCAN_INTERVAL, default=300): int,
+                }
+            ),
```

Comment:
You should set the schema to be a constant, and then maybe use `add_suggested_values`